frame/nfts/src/tests.rs

```diff
@@ -551,56 +551,116 @@ fn cancel_approval_works() {
 		assert_ok!(Nfts::force_create(Origin::root(), 0, 1, true));
 		assert_ok!(Nfts::mint(Origin::signed(1), 0, 42, 2));
 
-		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 3));
+		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 3, None));
 		assert_noop!(
-			Nfts::cancel_approval(Origin::signed(2), 1, 42, None),
+			Nfts::cancel_approval(Origin::signed(2), 1, 42, 3),
 			Error::<Test>::UnknownCollection
 		);
 		assert_noop!(
-			Nfts::cancel_approval(Origin::signed(2), 0, 43, None),
+			Nfts::cancel_approval(Origin::signed(2), 0, 43, 3),
 			Error::<Test>::UnknownCollection
 		);
 		assert_noop!(
-			Nfts::cancel_approval(Origin::signed(3), 0, 42, None),
+			Nfts::cancel_approval(Origin::signed(3), 0, 42, 3),
 			Error::<Test>::NoPermission
 		);
 		assert_noop!(
-			Nfts::cancel_approval(Origin::signed(2), 0, 42, Some(4)),
-			Error::<Test>::WrongDelegate
+			Nfts::cancel_approval(Origin::signed(2), 0, 42, 4),
+			Error::<Test>::NotDelegate
 		);
 
-		assert_ok!(Nfts::cancel_approval(Origin::signed(2), 0, 42, Some(3)));
+		assert_ok!(Nfts::cancel_approval(Origin::signed(2), 0, 42, 3));
 		assert_noop!(
-			Nfts::cancel_approval(Origin::signed(2), 0, 42, None),
-			Error::<Test>::NoDelegate
+			Nfts::cancel_approval(Origin::signed(2), 0, 42, 3),
+			Error::<Test>::NotDelegate
 		);
 	});
 }
 
+#[test]
+fn approving_multiple_accounts_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Nfts::force_create(Origin::root(), 0, 1, true));
+		assert_ok!(Nfts::mint(Origin::signed(1), 0, 42, 2));
+
+		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 3, None));
+		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 4, None));
+		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 5, None));
```

Comment:
I would replace it with:
```rust
		let current_block = 1;
		System::set_block_number(current_block);
		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 3, None));
		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 4, None));
		assert_ok!(Nfts::approve_transfer(Origin::signed(2), 0, 42, 5, Some(2)));
		assert_eq!(approvals(0, 42), vec![(3, None), (4, None), (5, Some(current_block + 2))]);
```
to validate the approvals were set correctly.

Also you'll need to add:
```rust
fn approvals(collection_id: u32, item_id: u32) -> Vec<(u64, Option<u64>)> {
	let item = Item::<Test>::get(collection_id, item_id).unwrap();
	let s: Vec<_> = item.approvals.into_iter().collect();
	s
}
```
after `fn attributes(collection: u32)...`